pfsenseapi/interface.go

```diff
@@ -242,10 +268,26 @@ func (s InterfaceService) DeleteVLAN(ctx context.Context, idToDelete int) error
 	return nil
 }
 
+func (s InterfaceService) getVLANIndex(ctx context.Context, vlanIf string) (int, error) {
```

Comment:
Rather than accepting a vlanIf and performing another call to get the index, I am wondering if it would be better to continue to accept an ID for the InterfaceService for consistency. 

Is there a specific reason you need this to be in the api client?

Reply:
The API client should abstract all the API weirdness, this is API weirdness

Reply:
Works for me. We can use this pattern for new update methods as well.